### ISSUE TITLE
c++ version of this crate

### ISSUE THREAD

**Issue by porky11** [WaDelma/galgebra#3]
You may look at this if you don't really understand.
https://github.com/wolftype/versor
I fear, that in current rust, something like this is not possible, because of the many `constexpr` functions. I'm not sure yet if this can be fixed using typenum and generic arrays, or if we have to wait for rust getting const generics (which already seem to be accepted).
I would like to have something like versor in rust, but with more explicit typing unlike the c++ version.
Maybe you already know this, because you tried to implement this, and wanted to make something similar. But if not this could be a good reference.

**Comment by WaDelma** [comment]
I was going to try and make geometric algebra library [galgebra](https://github.com/WaDelma/galgebra), but I got stuck as `typenum` isn't perfect and extremely hard to work with. So I am pretty much waiting for integer generics before I start with it again.

**Comment by WaDelma** [comment]
Yeah, I am pretty much waiting for const generics at this point as `typenum`/`generic-array` gets extremely flaky if I do constant to some power and I don't think I can get around that.

Good to have something that I can reference to!
